### ISSUE TITLE
Pdf file transform to image have a black block

### ISSUE THREAD

**Issue by Agoin-max** [pymupdf/PyMuPDF#3624]
### Description of the bug

Pdf file transform to image have a black block
[89b59dbfae5e4d1d92596418e9585a10.pdf](https://github.com/user-attachments/files/15984196/89b59dbfae5e4d1d92596418e9585a10.pdf)

### How to reproduce the bug
```
def pdf2png_with_pymupdf(pdf_data: Union[bytes, str], matrix: int = 2):
    """转换图片."""
    images: List[Image.Image] = []
    path = tempfile.mkdtemp()
    path_ = Path(path)

    try:
        if isinstance(pdf_data, bytes):
            pdf_path = str(path_.joinpath("mypdf.pdf"))
            with open(pdf_path, "wb") as fs:
                fs.write(pdf_data)
        else:
            pdf_path = pdf_data

        doc = fitz.open(pdf_path)
        for page_index in range(len(doc)):
            page = doc.load_page(page_index)
            pix = page.get_pixmap(matrix=fitz.Matrix(matrix, matrix))  # type: ignore
            img = Image.frombytes("RGB", [pix.width, pix.height], pix.samples)  # type: ignore
            images.append(img)
        doc.close()
    finally:
        delete_temp_directory(path)
```
       this is my code

### PyMuPDF version

1.24.6

### Operating system

Windows

### Python version

3.9

**Comment by JorjMcKie** [comment]
This is an issue in MuPDF.
Created an item in its tracker: https://bugs.ghostscript.com/show_bug.cgi?id=707845

**Comment by sebras** [comment]
@JorjMcKie According to the upstream bug report this has been fixed, and thus this bug could be marked "fix developed"?

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.24.11.

**Comment by Agoin-max** [comment]
> Fixed in 1.24.11.

Thanks
